test/fakeintake/server/server_test.go

```diff
@@ -28,7 +29,14 @@ import (
 )
 
 func TestServer(t *testing.T) {
+	for _, driver := range []string{"memory", "sql"} {
+		os.Setenv("STORAGE_DRIVER", driver)
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
`Setenv` should be avoided as it is not thread safe, prefer passing a flag to test this https://www.evanjones.ca/setenv-is-not-thread-safe.html

Reply:
It depends on the library implementation. Apparently the new one I just bumped supports concurrent accesses.

Reply:
I thought so as well. Also using a global state is not the best option. Today it's not an issue because we don't have concurrent threads accessing it but it could become one if new tests are added using `NewStore()` so indeed it's better to remove it.
I added an option in the last commit